### PR TITLE
Case 22483: Disable the use of tbb::parallel_for in Blender

### PR DESCRIPTION
**Pull request by jherico**
[22483](https://highfidelity.manuscript.com/f/cases/22483/tbb-causing-significant-cpu-overhead):  The current public build in HQ has an unreasonable amount of CPU usage, apparently regardless of the number of CPU cores available on a machine.  

Testing determined that the bulk of this appears to be coming from the tbb (threading building blocks) library, specifically the use of `tbb::parallel_for` in the `Blender::run` method.  Removing the use of tbb here drastically reduced the CPU usage of interface in local tests while connected to hq.

The basic problem is twofold.  First, the syntax of the existing code doesn't do any _chunking_, i.e. if you have 100 work items it creates 100 individual tasks that each do 1 item, rather than say, 5 tasks that each do 20 items.  Second the `Blender` task has already been dispatched to another thread, and it's possible that multiple `Blender` tasks may run concurrently.  Asking each of them to then use _all the cores_ on the system seems to negate the point of pushing Blender to it's own thread.  Also, the normal packing task is excessively simple and the 

The two issues mean that each run of the Blender task may cause hundreds of tasks to be generated, maximizing the cost of the tbb overhead far beyond any potential savings.  

In attempting to address this I both tried increasing the number of items per _chunk_ for the tbb approach, as well as trying an approach using OpenMP to see if it had better performance.  The chunk size seemed to have no effect, and the OpenMP approach made things much worse.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19234/HighFidelity-Beta-PR15517-122eafb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19234/HighFidelity-Quest-Beta-PR15517-122eafb.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19234/HighFidelity-Beta-PR15517-122eafb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19234/HighFidelity-Beta-Interface-PR15517-122eafb.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19242/HighFidelity-Beta-PR15517-3c63743.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19242/HighFidelity-Quest-Beta-PR15517-3c63743.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19248/HighFidelity-Beta-PR15517-2b1267f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19248/HighFidelity-Quest-Beta-PR15517-2b1267f.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19248/HighFidelity-Beta-PR15517-2b1267f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19248/HighFidelity-Beta-Interface-PR15517-2b1267f.exe

**Comment by jherico** [comment]
build this please

File: libraries/render-utils/src/Model.cpp [hunk 1]
```diff
@@ -1741,15 +1741,23 @@ Blender::Blender(ModelPointer model, int blendNumber, const Geometry::WeakPointe
     _blendshapeCoefficients(blendshapeCoefficients) {
 }
 
+#define BLENDER_USE_NONE 0
+#define BLENDER_USE_TBB 1
+#define BLENDER_TBB_CHUNK_SIZE 512
+#define BLENDER_USE_OPENMP 2
```
Comment (SamGondelman) [inline libraries/render-utils/src/Model.cpp]: If we know TBB and OpenMP make things worse, we should just get rid of these and not clutter up this horrible file.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19254/HighFidelity-Beta-PR15517-2b1267f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19254/HighFidelity-Beta-Interface-PR15517-2b1267f.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19254/HighFidelity-Beta-PR15517-2b1267f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19254/HighFidelity-Quest-Beta-PR15517-2b1267f.apk)

File: libraries/render-utils/src/Model.cpp (hunk 1 above)
Comment (jherico) [reply libraries/render-utils/src/Model.cpp]: I wanted something that would be easy to revert or experiment with, but I guess you're right.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19256/HighFidelity-Beta-PR15517-17632e3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19256/HighFidelity-Beta-Interface-PR15517-17632e3.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19256/HighFidelity-Beta-PR15517-17632e3.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19256/HighFidelity-Quest-Beta-PR15517-17632e3.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19256/HighFidelity-Beta-PR15517-17632e3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19256/HighFidelity-Beta-Interface-PR15517-17632e3.exe

**Pull request merged**